### PR TITLE
Setup Typescript in Theseus

### PR DESCRIPTION
**Pull request by ferothefox**

### CODE REVIEW

**Comment by brawaru** [comment]
You probably want to include [`vue-tsc`](https://www.npmjs.com/package/vue-tsc) to type check all files including Vue SFCs (the script would be `vue-tsc --noEmit`).

**Comment by ferothefox** [comment]
> You probably want to include [`vue-tsc`](https://www.npmjs.com/package/vue-tsc) to type check all files including Vue SFCs (the script would be `vue-tsc --noEmit`).

I'll get around to doing this for sure - was mainly focused on getting a dirty setup out real quick.

**Review by Geometrically — APPROVED** [review]

**Review by Prospector — APPROVED** [review]

**Review by Prospector — APPROVED** [review]

File: apps/app-frontend/tsconfig.json [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-    "moduleResolution": "NodeNext",
-    "baseUrl": ".",
-    "paths": {
-      "@/*": ["src/*"]
-    },
-    "target": "ESNext"
+    "allowJs": true,
+    "target": "ESNext",
+    "module": "ESNext",
+    "moduleResolution": "node",
```
Comment (brawaru) [inline apps/app-frontend/tsconfig.json:6]: This should be set to `"bundler"` since we're using Vite.

File: apps/app-frontend/tsconfig.json [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+    "types": ["vite/client"],
+    "strict": true,
+    "noEmit": true,
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "isolatedModules": true
   },
+  "include": ["src/**/*.ts", "src/**/*.d.ts", "src/**/*.vue"],
```
Comment (brawaru) [inline apps/app-frontend/tsconfig.json:14]: This can be simplified.
```suggestion
  "include": ["src"],
```

**Review by brawaru — CHANGES_REQUESTED** [review]
Honestly, I think it would be better to simply take TypeScript configs from Vite's own template — https://github.com/vitejs/vite/tree/main/packages/create-vite/template-vue-ts. They even use compositing to make TypeScript check `vite.config.ts` properly.

**Comment by ferothefox** [comment]
Succeeded by #2364

**Pull request closed without merging**